src/NHibernate/Hql/Ast/ANTLR/Tree/BinaryArithmeticOperatorNode.cs

```diff
@@ -32,50 +32,50 @@ public void Initialize()
 			IType lhType = (lhs is SqlNode) ? ((SqlNode)lhs).DataType : null;
 			IType rhType = (rhs is SqlNode) ? ((SqlNode)rhs).DataType : null;
 
-			if (lhs is IExpectedTypeAwareNode && rhType != null)
+			TrySetExpectedType(lhs, rhType, true);
+			TrySetExpectedType(rhs, lhType, false);
```

Comment:
The previous code would have set rhs expected type only if lhs was not a type aware node or rhType was null.
Was this a bug fixed by the way? Or is this an overlooked change?

Reply:
Only a `ExpectedType != null` check was added and the `rhs` check was modified to check for `IExpectedTypeAwareNode` instead `ParameterNode`, other changes are code cleanup. 
`ParameterNode` implements `IExpectedTypeAwareNode` and checking for `ParameterNode` instead of `IExpectedTypeAwareNode` would ommit `BooleanLiteralNode`, which doesn't seem right to me. If it is copied for `lhs` it should be also copied for `rhs`. So it was intentional, but I forgot to mention it.